subscriptions/providers/apple_in_app/__init__.py

```diff
@@ -260,13 +290,13 @@ def _handle_subscription_change(self, notification: AppStoreNotification) -> Non
         # 2. We add a new subscription with a new plan.
 
         latest_payment = self._get_latest_transaction(transaction_info.original_transaction_id)
-        # Purchase date is the time when we should stop the previous subscription.
-        previous_payment_end_time = transaction_info.purchase_date
+        assert latest_payment, f'Change received for {transaction_info=} where no payments exist.'
 
         # Ensuring that subscription ends earlier before making the payment end earlier.
-        latest_payment.subscription.end = previous_payment_end_time
+        latest_payment.subscription.end = timezone.now()
         latest_payment.subscription.save()
-        latest_payment.subscription_end = previous_payment_end_time
+        latest_payment.subscription_end = timezone.now()
```

Comment:
`latest_payment.subscription_end` will be > than `latest_payment.subscription.end` (bc calls to `timezone.now()` are subsequent) which will trigger extending `latest_payment.subscription.end` -> `latest_payment.subscription_end`. Just calculate `now_ = timezone.now()` and use it in both cases.